Downloader failing for ssh-based project-url

### How to reproduce

1. Pull and compile ORT. I spotted this issue first in revision e771a67, but the latest revision from main branch also affected. 
2. Make sure you have ssh key registered in GitHub and it's accessible in the same environment where you run ORT.
3. Run ORT downloader as following: 
```
/opt/ort/bin/ort --debug \
    download \
    --project-name "OSS_Review_Toolkit" \
    --project-url "git@github.com:jshttp/mime-types.git" \
    --vcs-type "git" \
    --vcs-revision "c77d6b02139075e4f63230a970411b3c92d066e2" \
    --vcs-path "" \
    -o project  
```

It will fail with error: 
```
09:04:12.304 [main] ERROR org.ossreviewtoolkit.plugins.commands.downloader.DownloaderCommand - The following failure(s) occurred:
DownloadException: Download failed for 'Downloader::OSS_Review_Toolkit:'.
Suppressed: DownloadException: Git failed to get revisions from URL 'ssh://github.com/jshttp/mime-types.git'.
    Caused by: DownloadException: Unable to determine a revision to checkout.
        Suppressed: IOException: Cannot guess a revision name from a blank version.
        Suppressed: IOException: Unable to list the remote branches.
            Caused by: TransportException: ssh://github.com/jshttp/mime-types.git: remote hung up unexpectedly
                Caused by: TransportException: ssh://github.com/jshttp/mime-types.git: remote hung up unexpectedly
                    Caused by: NoSuchMethodError: 'java.lang.Object org.apache.sshd.client.future.ConnectFuture.verify(org.apache.sshd.common.future.CancelOption[])'Suppressed: DownloadException: No source artifact URL provided. 
```

### Observations

- Running the same command with [http](https://github.com/jshttp/mime-types.git) project-url instead of ssh passes without errors.
- Previous versions (c05ec08 and older) doesn't have this issue.
- Pure git (clone, fetch, etc) works fine on the same environment over ssh with the same key.
- Might be related to jgit [update](https://github.com/oss-review-toolkit/ort/commit/115f24149b0b720f4550009d3316653c3940d1c6) from 6.5.0.202303070854-r to 6.6.0.202305301015-r as it includes dependency update that might cause changes in ssh behavior - [581770](https://eclip.se/581770) Apache mina-sshd to 2.10.0. (Just a guess, I don't see related bug's or features in release notes).

[downloader_debug.log](https://github.com/oss-review-toolkit/ort/files/11832665/downloader_debug.log)

--- a/downloader/build.gradle.kts
+++ b/downloader/build.gradle.kts
@@ -29,7 +29,12 @@ dependencies {
 
     implementation(libs.jgit)
     implementation(libs.jgitSshApacheAgent)
-    implementation(libs.svnkit)
+    implementation(libs.svnkit) {
+        exclude(group = "org.apache.sshd", module = "sshd-core")
+            .because("it is included in JGit's sshd-osgi dependency")
+        exclude(group = "org.apache.sshd", module = "sshd-common")
+            .because("it is included in JGit's sshd-osgi dependency")
+    }
 
     testImplementation(libs.mockk)
 }

